buffer/llbuffer.go

```diff
@@ -0,0 +1,107 @@
+package buffer
+
+import (
+	"container/list"
+	"io"
+	"sync"
+)
+
+type FileChunk struct {
+	Start   int64
+	Content []byte
+}
+
+type OrderedBuffer struct {
+	w             io.Writer
+	totalExpected int64
+	written       int64
+	list          *list.List
+	mu            *sync.Mutex
+	Queued        int
+}
+
+func NewOrderedBuffer(expectedBytes int64, w io.Writer) *OrderedBuffer {
+	return &OrderedBuffer{
+		totalExpected: expectedBytes,
```

Comment:
`totalExpected` is unused, hence `expectedBytes` parameter is unused, hence extra `headobject` call in https://github.com/peak/s5cmd/pull/593/files#diff-015e2d270a318042cca60407874486ab8b45b1c3f60b81e9e1b3563ae5143b95R115 is unnecessary.

Reply:
You are right that `totalExpected` and `expectedBytes` are unnecessary, but the `HeadObject` is useful for correct error messages. When an object is not found, the head returns the humanly formatted error message while if we handle during the `Get` call, there are status code, host ID, etc. Here are the comparison for both: 
`HeadObject` error:
```
{"operation":"cat","command":"cat s3://test-cat-s-3-object-failcat-non-existent-r-ah5j9t5/prefix/file.txt","error":"given object s3://test-cat-s-3-object-failcat-non-existent-r-ah5j9t5/prefix/file.txt not found"}
```
`Get` error:
```
{"operation":"cat","command":"cat s3://test-cat-s-3-object-failcat-non-existent-r-rvvlx5y/prefix/file.txt","error":"NoSuchKey:  status code: 404, request id: 0000000000000003, host id: MDAwMDAwMDAwMDAwMDAwMzAwMDAwMDAwMDAwMDAwMDMwMDAwMDAwMDAwMDAwMDAzMDAwMDAwMDAwMDAwMDAwMw=="}
```